### PR TITLE
Updated script to check inconsistent pandas namespace

### PR DESCRIPTION
**Pull request by mohdkashif93**
- [ ] closes #37838 
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

Updates the script for inconsistent namespace usage. To be merged after all the errors are fixed with the updated script. The script is described here : https://github.com/MarcoGorelli/PyDataGlobal2020-sprint/issues/1 by @MarcoGorelli

### CODE REVIEW

**Comment by jreback** [comment]
we will likely need to merge master to make sure after some other merges (but wait till this is green)

**Comment by jreback** [comment]
can you merge master and see if you can fixup and remaining issues ( pre-commits are failing). ping on green.

**Comment by MarcoGorelli** [comment]
this is pending on #37852 , once that's in I'd expect this to pass

**Comment by MarcoGorelli** [comment]
Hi @mohdkashif93 - can you try fetching and merging upstream/master now please and see if this passes? If not, do you want to fix up the remaining files as well?

**Comment by jreback** [comment]
yep this needs a rebase and merge on green.

**Comment by mohdkashif93** [comment]
Yes @MarcoGorelli , I will rebase and check if there are any pending errors with the script 😄

File: scripts/check_for_inconsistent_pandas_namespace.py [hunk 1]
```diff
@@ -17,28 +17,17 @@
...
-        pd\.{class_name}\(  # only match e.g. pd.DataFrame
+        pd\.\2\(  # only match e.g. pd.DataFrame
     )|
     (
-        pd\.{class_name}\(  # only match e.g. pd.DataFrame
+        pd\.([A-Z]\w+)\(  # only match e.g. pd.DataFrame
         .*                  # match anything
         (?<!pd\.)(?<!\w)    # check class_name start with pd. or character
-        {class_name}\(      # match DataFrame but not pd.DataFrame or tm.makeDataFrame
+        \4\(      # match DataFrame but not pd.DataFrame or tm.makeDataFrame
```
Comment (MarcoGorelli) [inline scripts/check_for_inconsistent_pandas_namespace.py:28]: also, when you do the rebase, can you move the comments forwards so they all aligned again?

**Comment by jreback** [comment]
can you merge master

**Comment by MarcoGorelli** [comment]
@jreback I merged master and fixed up the few remaining files - good to go?

**Comment by jreback** [comment]
great thanks @mohdkashif93

**Pull request merged**